xbmc/filesystem/AudioBookFileDirectory.cpp

```diff
@@ -75,6 +75,53 @@ bool CAudioBookFileDirectory::GetDirectory(const CURL& url,
   if (m_fctx->nb_chapters > 1)
     thumb = IMAGE_FILES::URLFromFile(url.Get(), "music");
 
+  //Look for any embedded art
+  size_t size = 0;
+  uint8_t* pic{};
+  std::string type;
+
+  for (size_t i = 0; i < m_fctx->nb_streams; ++i)
+  {
+    if ((m_fctx->streams[i]->disposition & AV_DISPOSITION_ATTACHED_PIC) == 0)
+      continue;
+    AVDictionaryEntry* avtag;
+    avtag = av_dict_get(m_fctx->streams[i]->metadata, "filename", nullptr, AV_DICT_IGNORE_SUFFIX);
```

Comment:
Declare and init in one go: `AVDictionaryEntry* avtag{av_dict_get(m_fc...};`

Reply:
Decided to init to nullptr in case we don't get a tag back. That OK ?

Reply:
If the function returns a nullptr avtag gets inited with this value. Absolutely no need for two steps here.